### DISCUSSION TITLE
DHOTools for the Sparrow Extended UI and the RIGOL DHO 800/900 series oscilloscopes

### DISCUSSION THREAD

**Discussion by mriscoc** [mriscoc/Sparrow-Extended-UI-for-RIGOL-DHO800_DHO900#1] [General]
DHOTools is an application designed to expand the capabilities of RIGOL oscilloscopes of the DHO800 and DHO900 series. It allows you to perform advanced analyses, such as FFT averaging, long-term data logging, and arbitrary zoom of a signal, from a PC with Windows or Linux, directly on the oscilloscope or from your mobile device (cell phone or tablet) via a network connection.

Release page: https://github.com/mriscoc/Sparrow-Extended-GUI-for-RIGOL-DHO800_DHO900/releases/tag/DHOTools

https://github.com/user-attachments/assets/c93b5b5a-fd92-4610-a4f2-295620d3c0b6

![DHO-datalogger](https://github.com/user-attachments/assets/2bf18a5c-d025-486b-b651-26f75eab5eee)

![DHO-freezoom](https://github.com/user-attachments/assets/f945e1d7-06cf-42e6-8083-bcb8ea45dfb5)

<img width="1021" height="637" alt="Captura de pantalla 2026-09-08 234247" src="https://github.com/user-attachments/assets/7c88870e-2181-4ec9-92f4-04a7db8bbfa4" />

<img width="1021" height="637" alt="Captura de pantalla 2026-09-08 234218" src="https://github.com/user-attachments/assets/bb93b1ab-8b1c-43d2-ac54-7187498d4c70" />

## How to install in the DHO
Connect the Oscilloscope to the network and check its current IP. Copy the apk file to android-sdk\platform-tools on your PC, then execute this command inside of the platform-tools folder:
```terminal
> adb connect [oscilloscope IP]:55555
connected to [oscilloscope IP]:55555
```

Install the utility using:
```terminal
> adb install -r -g DHOTools_0.3.4.apk
Performing Streamed Install
Success
```

How to uninstall:
```shell
> adb uninstall org.mriscoc.dhotools
```

Start the application from the launcher or by using adb:
```terminal
> adb shell monkey -p org.mriscoc.dhotools -c android.intent.category.LAUNCHER 1
Events injected: 1
## Network stats: elapsed time=36ms (0ms mobile, 0ms wifi, 36ms not connected)
```

DHOtools can be also launched from the [DHOActions panel](https://github.com/mriscoc/Sparrow-Extended-GUI-for-RIGOL-DHO800_DHO900/discussions/9#discussioncomment-12303439)

![DHOactions](https://github.com/user-attachments/assets/07d342a6-bc9b-4090-ae44-8e642dce518e)

> [!NOTE]
> This version only works if you have previously installed the  [Sparrow Extended](https://www.patreon.com/posts/141472182) update.

## Connecting a DGE20XX to a DHO800/900 series Rigol oscilloscopes

> [!NOTE]
> Due to poor performance and limited usability on Android versions, it is advisable to run the frequency response page of the DHOTools remotely on a PC. 

The firmware built for the DHO devices does not include the correct library to enable a DGE20XX device to connect via USB. Therefore, a library has been compiled for this platform and needs to be installed in a system folder. 
Decompress the library from the ZIP file:

[libusb-1.0.so.zip](https://github.com/user-attachments/files/28965286/libusb-1.0.so.zip)

Install this library in the DHO by using ADB:
```bash
  adb root
  adb remount
  adb push libusb-1.0.so /system/lib64/
  adb shell chmod 644 /system/lib64/libusb-1.0.so
```
Connect the DGE20XX to the USB port and reboot the DHO. Once the device has rebooted, check if the waveform generator was detected:
```bash
  adb shell su -c lsusb
```
PID:VID detected:
```bash
Bus 002 Device 003: ID 5345:1234
```
Now change the permissions for the device:
```bash
  adb shell su -c "chmod 666 /dev/bus/usb/*/*"
```
To avoid changing the permissions with each reboot, add the 'chmod' command to the startup script at '/Rigol/shell/start_rigol_app.sh'. Then install the DHOTools APK. The first launch will abort due to the time it takes for the application to decompress.

**Comment by RAPosthumus** [comment]
Hi my name is Rense. 
I have a DHO924S (and a SDS804X HD) to play with.
Thanks for helping me out with the zone-edge app.

Four wishes:
Upon connecting to the data server, check which (first) math channel the FFT is displayed on and use that as the base for displaying it.
-could the app intercept the 'quick' button from the scope and execute that action?
-make it possible to use a logarithmic axis
-a possibility to view frequency and the phase information of a signal.

**Reply by mriscoc** [reply]
Original plot  
![image](https://github.com/mriscoc/RIGOL_DHO800_DHO900_GUI/assets/2745567/26137aa9-9b8e-4e86-bb08-027103ebd1c1)
  

This is the result for semi logarithmic X axis  
![image](https://github.com/mriscoc/RIGOL_DHO800_DHO900_GUI/assets/2745567/44965244-b3dd-485c-a956-591a4343ec5b)

**Reply by mriscoc** [reply]
So, at low frequencies there is a loss of resolution. It is acceptable for you?

**Comment by RAPosthumus** [comment]
First reaction: yes but we should quantify the loss.

Op do 11 apr. 2024 18:39 schreef Miguel Risco-Castillo <
***@***.***>:

**Reply by mriscoc** [reply]
I added button to enable xlog scale, I will release the new apk soon.

**Comment by RAPosthumus** [comment]
Awesome, thanks for the update!  Will test this in the weekend

Op do 18 apr. 2024 19:27 schreef Miguel Risco-Castillo <
***@***.***>:

**Reply by RAPosthumus** [reply]
Testing reveals some errors/room for enbhancements. See attached 5 test reports.
[01_generalremarks.pdf](https://github.com/mriscoc/RIGOL_DHO800_DHO900_GUI/files/15051758/01_generalremarks.pdf)
[02_differentvalues.pdf](https://github.com/mriscoc/RIGOL_DHO800_DHO900_GUI/files/15051759/02_differentvalues.pdf)
[03_missing peak.pdf](https://github.com/mriscoc/RIGOL_DHO800_DHO900_GUI/files/15051760/03_missing.peak.pdf)
[04_shifted startfreq.pdf](https://github.com/mriscoc/RIGOL_DHO800_DHO900_GUI/files/15051761/04_shifted.startfreq.pdf)
[05_missing_peakvalue.pdf](https://github.com/mriscoc/RIGOL_DHO800_DHO900_GUI/files/15051762/05_missing_peakvalue.pdf)

**Reply by mriscoc** [reply]
Thank you for your suggestions. I'm making some improvements to the tool

https://www.youtube.com/watch?v=NlYNN4ZO-7Y

**Comment by generalasde** [comment]
Hello.

I have installed the Windows version, but it doesn’t connect and I don’t know how to specify the DHO’s IP address. Also, when I launch the APK on the DHO, it opens in full screen and I don’t know how to make it use only half of the screen like in the examples you show.

**Reply by mriscoc** [reply]
DHOTools for Windows will automatically detect any DHO connected to the same subnetwork. You don't need to specify an IP address; just make sure that the DHO is connected to the network and has an IP address before launching DHOTools.
To open the DHOTools APK on Android in split-screen mode, you need an application that can enable this feature. I use Zonelauncher (https://play.google.com/store/apps/details?id=com.bialy.zonelauncher) in my examples, but any other similar application can be used.

**Reply by generalasde** [reply]
The DHO and the PC are on the same subnet, and I can connect using both DHOVision and the web server. I’m not sure what could be causing the issue that prevents me from connecting.

**Reply by mriscoc** [reply]
Ok, I confirm the bug in the 0.3.4 version of DHOTools, it is looking for a device in a fixed ip address

**Reply by mriscoc** [reply]
Please try with this version: https://github.com/mriscoc/Sparrow-Extended-GUI-for-RIGOL-DHO800_DHO900/releases/download/DHOTools/DHOTools_setup_0.3.5_WINx64.exe

**Reply by generalasde** [reply]
Thank you for the quick response. It works perfectly.

**Comment by mriscoc** [comment]
![DHOTools-WinX64](https://github.com/user-attachments/assets/22e68005-7cd8-40a3-8a55-96dbc95d6ee3)

DHOTools for Windows, updated to 0.3.6:
- THD button in the FFT screen.
